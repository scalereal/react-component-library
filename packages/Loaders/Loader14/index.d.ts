import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader14: React.FC<Props> & WithStyle;
export default Loader14;
