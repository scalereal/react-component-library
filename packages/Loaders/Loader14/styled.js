var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    border-right: .1em solid currentcolor;\n\tborder-radius: 100%;\n\tanimation: loader-37 800ms linear infinite;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\twidth: .8em;\n\t\theight: .8em;\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\ttop: calc(50% - .4em);\n\t\tleft: calc(50% - .4em);\n\t\tborder-left: .08em solid currentcolor;\n\t\tborder-radius: 100%;\n\t\tanimation: loader-37 400ms linear infinite reverse;\n\t}\n\t&:after {\n\t\twidth: .6em;\n\t\theight: .6em;\n\t\ttop: calc(50% - .3em);\n\t\tleft: calc(50% - .3em);\n\t\tborder: 0;\n\t\tborder-right: .05em solid currentcolor;\n\t\tanimation: none;\n\t}\n\n    @keyframes loader-37 {\n        from {\n            transform: rotate(360deg);\n        }\n        to {\n            transform: rotate(0deg);\n        }\n    }\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    border-right: .1em solid currentcolor;\n\tborder-radius: 100%;\n\tanimation: loader-37 800ms linear infinite;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\twidth: .8em;\n\t\theight: .8em;\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\ttop: calc(50% - .4em);\n\t\tleft: calc(50% - .4em);\n\t\tborder-left: .08em solid currentcolor;\n\t\tborder-radius: 100%;\n\t\tanimation: loader-37 400ms linear infinite reverse;\n\t}\n\t&:after {\n\t\twidth: .6em;\n\t\theight: .6em;\n\t\ttop: calc(50% - .3em);\n\t\tleft: calc(50% - .3em);\n\t\tborder: 0;\n\t\tborder-right: .05em solid currentcolor;\n\t\tanimation: none;\n\t}\n\n    @keyframes loader-37 {\n        from {\n            transform: rotate(360deg);\n        }\n        to {\n            transform: rotate(0deg);\n        }\n    }\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + Loader + "{\n                margin: 0 2px 2px 2px;\n            }\n           ";
        case "XS":
            return "font-size: 1rem;\n            " + Loader + "{\n                margin: 0 3px 3px 3px;\n            }\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            " + Loader + "{\n                margin: 0 5px 5px 5px;\n            }\n            ";
        case "M":
            return "font-size: 3rem;\n            " + Loader + "{\n                margin: 0 5px 5px 5px;\n            }\n            ";
        case "L":
            return "font-size: 5rem;\n            " + Loader + "{\n                margin: 0 10px 10px 10px;\n            }\n            ";
        case "XL":
            return "font-size: 6rem;\n            " + Loader + "{\n                margin: 0 10px 10px 10px;\n            }\n            ";
        default:
            return "font-size: 1rem;\n        " + Loader + "{\n            margin: 0 3px 3px 3px;\n        }\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.display ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
