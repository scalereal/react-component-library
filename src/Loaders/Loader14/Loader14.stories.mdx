import Loader14 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader14"
    component={Loader14}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader14 {...args} />
    </div>
);

<Preview>
    <Story name="Loader14" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader14 size="S" Color="green" />
        <Loader14 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader14 size="XXS" />
        <Loader14 size="XS" />
        <Loader14 size="S" />
        <Loader14 size="M" />
        <Loader14 size="L" />
        <Loader14 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader14" />
