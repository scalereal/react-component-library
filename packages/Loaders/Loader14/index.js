var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Loader, LoaderWrapper } from './styled';
var Loader14 = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(Loader, __assign({}, props))));
}));
Loader14.displayName = "Loader14";
Loader14.defaultProps = {
    size: 'S',
    display: true
};
export default Loader14;
