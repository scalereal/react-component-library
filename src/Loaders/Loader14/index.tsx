import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader14:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader14.displayName="Loader14";
Loader14.defaultProps = {
    size:'S',
    visible: true
};

export default Loader14;
