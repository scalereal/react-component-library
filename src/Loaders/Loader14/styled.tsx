import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    border-right: .1em solid currentcolor;
	border-radius: 100%;
	animation: loader-37 800ms linear infinite;
	&:before,
	&:after {
		content: '';
		width: .8em;
		height: .8em;
		display: block;
		position: absolute;
		top: calc(50% - .4em);
		left: calc(50% - .4em);
		border-left: .08em solid currentcolor;
		border-radius: 100%;
		animation: loader-37 400ms linear infinite reverse;
	}
	&:after {
		width: .6em;
		height: .6em;
		top: calc(50% - .3em);
		left: calc(50% - .3em);
		border: 0;
		border-right: .05em solid currentcolor;
		animation: none;
	}

    @keyframes loader-37 {
        from {
            transform: rotate(360deg);
        }
        to {
            transform: rotate(0deg);
        }
    }
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 0 2px 2px 2px;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 0 3px 3px 3px;
            }
            
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 0 5px 5px 5px;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 0 5px 5px 5px;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 0 10px 10px 10px;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 0 10px 10px 10px;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin: 0 3px 3px 3px;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
