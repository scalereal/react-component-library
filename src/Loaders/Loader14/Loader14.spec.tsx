import { render } from '@testing-library/react';
import * as React from 'react';
import Loader14 from '.';

describe('Loader14 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader14/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader14
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader14 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});