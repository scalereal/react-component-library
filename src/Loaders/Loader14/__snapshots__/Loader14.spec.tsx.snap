// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader14 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz llLtCb"
  >
    <div
      class="sc-bdfBwQ aQkpH"
    />
  </div>
</div>
`;

exports[`Loader14 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz XCGmd"
  >
    <div
      class="sc-bdfBwQ hlgVGO"
    />
  </div>
</div>
`;

exports[`Loader14 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz AylWc"
  >
    <div
      class="sc-bdfBwQ aQkpH"
    />
  </div>
</div>
`;
